staging/src/k8s.io/kubectl/pkg/cmd/config/set_credentials.go

```diff
@@ -306,6 +315,12 @@ func (o *setCredentialsOptions) modifyAuthInfo(existingAuthInfo clientcmdapi.Aut
 			modifiedAuthInfo.Exec.Args = o.execArgs
 		}
 
+		if o.execInteractiveMode.Provided() {
+			modifiedAuthInfo.Exec.InteractiveMode = clientcmdapi.ExecInteractiveMode(o.execInteractiveMode.Value())
+		}
+
+		modifiedAuthInfo.Exec.ProvideClusterInfo = o.execProvideClusterInfo
```

Comment:
shouldn't this be conditional on whether they provided the flag? otherwise an invocation that tries to modify an existing entry but doesn't set that flag will always set this field to `false`. The boolean isn't enough to tell if they set the value or not

Reply:
I've found `Tristate` type to manage boolean types as exactly you suggested and updated the code with that.